docs/Announcement-Post.md

```diff
@@ -0,0 +1,269 @@
+      updates, opt_state = optimizer.update(grads, opt_state, params=model)
+      model = eqx.apply_updates(model, updates)
+      return loss, model, opt_state
+```
+
+Under the hood, all we do is use Jax's `pjit` operator to shard the parameters and optimizer states along the axis
+you specify. We then use `hax.axis_mapping` to change the preferred mapping for computing the loss and gradients. Jax
+handles most of the heavy lifting, except Linear layers need a bit of handholding, which Haliax provides.
+
+Tensor Parallelism works in a broadly similar manner, with you specifying how "wide" you want model parallelism to be
+and specifying which axes of your model should be sharded. (For a transformer, this would typically be the Head
+and MLP axes.)
+
+In our experiments, we have found that, for models of the sizes that we can reasonably train with our resources, FSDP
+is more than enough to get good scaling, especially on TPU. In practice, with FSDP, we haven't found tensor parallelism
+to be useful for training transformers to at least 20B parameters. (Tensor parallelism is useful for inference, but
+Levanter's focus is on training.)
+
+For more details, please see the [Architectural Overview](XXX) in the Levanter documentation.
+
+#### Bitwise Reproducibility
+
+One of the benefits of Jax is that it offers strong guarantees for reproducibility. In particular, Jax's fine-grained
+control over random number generation makes it easy to ensure bitwise reproducibility, especially when using TPUs.
+Levanter takes advantage of this to offer bitwise reproducibility for training runs, even after preemption. As an example,
+here is a screenshot of a training run being resumed multiple times, even on different TPU pod slices:
+
+![plot showing bitwise reproducibility with four training runs superimposed with the exact same loss curve](figures/bitwise_repro_curve.png)
+
+The fact that you can't make out the different lines is the point: the training runs are bitwise identical,
+a huge advantage for debugging and reproducibility.
+
+#### Token Probability Visualization
+
+As we've been working with teams who are building models with their own domain-specific models, data preparation has
+been a significant challenge. In particular, it can be difficult to identify issues in the data that are causing
+particularly low loss. As a small step towards addressing this, we have added a feature to Levanter that allows you
+to visualize the probability of each token in a sample of the validation set during training. For novel data sets,
+this has allowed us to identify, e.g., highly but not perfectly redundant data (what we call "madlib duplicates") or other
+issues that might be causing abnormally low loss. We've also used it to qualitative assess how alternative architectures
+learn differently from Transformers.
+
+Here is an example of the token probability visualization in action:
+
+XXX legal data?
```

Comment:
Feel free to use example from backpack if you need: https://wandb.ai/stanford-mercury/levanter/runs/eiv0evi3kv93483620128070.0002?workspace=user-ivanzhou
![CleanShot 2023-05-08 at 19 50 44](https://user-images.githubusercontent.com/8097756/236981593-d22db379-4245-46ec-88ef-7febd462f22f.png)

Reply:
thank you!